.github/workflows/rust.yml

```diff
@@ -72,7 +72,7 @@ jobs:
       - uses: actions-rs/clippy-check@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
-          args: --no-default-features --features "clock" --all-targets # tests need the clock feature
+          args: --no-default-features --features=clock,serde --all-targets # tests need the clock feature
```

Comment:
I don't think we should change the CI like this; it should be possible to use the library as no_std, without wanting `serde`. There are currently real errors to fix in the `no_std` case (see previous review).

Should we add a `cargo check --no-default-features` to the CI? Or is `clippy` a superset of `check`?

Reply:
> Should we add a cargo check --no-default-features to the CI? Or is clippy a superset of check?

Good point! I think a `clippy --all` should cover everything and therefore there is no need to run clippy with `--no-default-features` - what we really need is just a `check --no-default-features`. WDYT?

> it should be possible to use the library as no_std, without wanting serde

The ICS20 impl relies on serde for correctness (to properly deal with `PacketData` and `Acknowledgement`) in the module callbacks such as [on_recv_packet](https://github.com/cosmos/ibc-rs/blob/main/crates/ibc/src/applications/transfer/context.rs#L263). Should we also feature-gate them? (Does it make sense to feature gate individual callbacks or should we simply feature gate the whole app itself?)

Reply:
Note that encoding to JSON is required by the [spec](https://github.com/cosmos/ibc/blob/main/spec/app/ics-020-fungible-token-transfer/README.md#data-structures) ->
> Note that both the FungibleTokenPacketData as well as FungibleTokenPacketAcknowledgement must be JSON-encoded (not Protobuf encoded) when they serialized into packet data. Also note that uint256 is string encoded when converted to JSON, but must be a valid decimal number of the form [0-9]+.

Reply:
Right... Here is my solution: [a2a3a3f](https://github.com/cosmos/ibc-rs/pull/331/commits/a2a3a3fa903a83f35fb581bd4c594b89b824c98a). Basically it puts the `transfer` application behind the `serde` feature. WDYT?